src/Extensions/Hosting.Services.Web/Middlewares/ObsoleteCorrelationHeadersMiddleware.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	/// <summary>
+	/// Adds support of old correlation tracking logic that might be used by other services
+	/// </summary>
+	/// <remarks>
+	/// This middleware should be removed after all services moved to use <see cref="Activity"/>
+	/// </remarks>
+	[Obsolete("Use it only if you need to comunicate with services that use old correlation", false)]
+	internal class ObsoleteCorrelationHeadersMiddleware
+	{
+		/// <summary>
+		/// Invoke middleware
+		/// </summary>
+		public Task InvokeAsync(HttpContext context)
+		{
+			ExtractCorrelationFromRequest(context.Request);
+			context.Response.OnStarting(SetCorrelationHeadersToResponce, context.Response);
+			return Task.CompletedTask;
+		}
+
+		private Task SetCorrelationHeadersToResponce(object state)
+		{
+			HttpResponse response = (HttpResponse)state;
+			Guid? oldCorrelation = Activity.Current?.GetObsoleteCorrelationId();
+			if (oldCorrelation.HasValue)
+			{
+				response.Headers.Add(CorrelationHeader, oldCorrelation.ToString());
+			}
+
+			return Task.CompletedTask;
+		}
+
+		private void ExtractCorrelationFromRequest(HttpRequest request)
+		{
+			Activity? activity = Activity.Current;
+			if (activity == null)
+			{
+				return;
+			}
+
+			Guid? oldCorrelation = ExtractCorrelationId(request);
+			if (oldCorrelation.HasValue)
+			{
+				activity.SetObsoleteCorrelationId(oldCorrelation.Value);
+			}
+
+			IQueryCollection dataSources = request.Query;
+
+			uint? transactionId = ParseUint(ExtractParameter(dataSources, s_transactionsNames));
+			if (transactionId.HasValue)
+			{
+				activity.SetObsoleteTransactionId(transactionId.Value);
+			}
+			else
+			{
+				return;
+			}
```

Comment:
no need for return

Reply:
Thank you, it was for another code that was deleted :)